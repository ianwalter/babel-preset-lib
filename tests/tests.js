const { test } = require('@ianwalter/bff')
const { transformSync } = require('@babel/core')
const preset = require('..')

test('basic config', ({ expect }) => {
  const src = `
    export default function out (...args) {
      return { count: args.length, ...args }
    }
  `
  const { code } = transformSync(src, { presets: [preset] })
  expect(code).toMatchSnapshot()
})
