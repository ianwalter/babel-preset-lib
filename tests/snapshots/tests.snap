// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic config 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime-corejs3/helpers/interopRequireDefault\\");

var _Object$defineProperty = require(\\"@babel/runtime-corejs3/core-js-stable/object/define-property\\");

_Object$defineProperty(exports, \\"__esModule\\", {
  value: true
});

exports.default = out;

var _objectSpread2 = _interopRequireDefault(require(\\"@babel/runtime-corejs3/helpers/objectSpread\\"));

function out() {
  for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
    args[_key] = arguments[_key];
  }

  return (0, _objectSpread2.default)({
    count: args.length
  }, args);
}"
`;
